### PR TITLE
Drop legacy polar

### PR DESCRIPTION
**Pull request by archmoj**
Addresses #5395,
This PR drops legacy `polar` and `area` trace.

@plotly/plotly_js

### CODE REVIEW

File: test/jasmine/tests/mock_test.js [hunk 1]
```diff
@@ -6,7 +6,6 @@ var list = [
     '4',
     '5',
     '6',
-    '7',
```
Comment (alexcjohnson) [inline test/jasmine/tests/mock_test.js:9]: And delete 6 and 8 (and their commented-out lines below), right?

Comment (alexcjohnson) [reply test/jasmine/tests/mock_test.js:9]: Should verify that all the `@noCI` tests still run before we merge this.

Comment (archmoj) [reply test/jasmine/tests/mock_test.js:9]: Good eye.
Done in https://github.com/plotly/plotly.js/pull/5399/commits/de6ca3982088abada0920e96c83063dc6dc1f82b.
And validated before submitting the PR.

Comment (alexcjohnson) [reply test/jasmine/tests/mock_test.js:9]: OIC, this line will skip missing figures
https://github.com/plotly/plotly.js/blob/1529f3b0304272a405cfdfac32e627487e7b5b0b/test/jasmine/tests/mock_test.js#L2178
That seems a little dangerous, but doesn't need to be addressed now.

**Review by alexcjohnson — APPROVED** [review]
💃 that's a nice big 🔪 !!

**Pull request merged**